packages/client/hmi-client/src/services/document-assets.ts

```diff
@@ -209,16 +211,25 @@ async function createDocumentFromXDD(
 	projectId: string
 ): Promise<AddDocumentAssetFromXDDResponse | null> {
 	if (!document || !projectId) return null;
-	const response = await API.post(`/document-asset/create-document-from-xdd`, {
-		document,
-		projectId
-	});
+	const response = await API.post<AddDocumentAssetFromXDDResponse>(
+		`/document-asset/create-document-from-xdd`,
+		{
+			document,
+			projectId
+		}
+	);
 
 	if (!response || response.status >= 400) {
 		logger.error('Error upload file from doi');
 		return null;
 	}
 
+	if (response.data.extractionJobId) {
+		const result = await fetchExtraction(response.data.extractionJobId);
+		if (result.state === PollerState.Done) {
+			modelCard(response.data.documentAssetId);
+		}
+	}
```

Comment:
This should be done in the controller of the `/document-asset/create-document-from-xdd`.
But I guess we have no way to know that it was done don't we?

Reply:
Everything should probably be done in the backend, Im not a fan of waiting for extractions/model card generation in the front end.  But I think that would require a pretty big refactor